src/inmanta/module.py

```diff
@@ -1750,6 +1915,38 @@ def sorted_modules(self) -> list:
 
         return mod_list
 
+    def add_module_requirement(self, requirement: InmantaModuleRequirement, add_as_v1_module: bool) -> None:
+        # Add requirement to metadata file
+        YamlMetadataFileWithRequiresField.add_module_requirement_and_write(
+            self, requirement, ignore_version_constraint=not add_as_v1_module
+        )
+        # Refresh in-memory metadata
+        with open(self.get_metadata_file_path(), "r", encoding="utf-8") as fd:
+            self._metadata = ProjectMetadata.parse(fd)
+        # Update requirements.txt file
+        requirements_txt_file_path = os.path.join(self._path, "requirements.txt")
+        if not add_as_v1_module:
+            requirements_txt_file = RequirementsTxtFile(requirements_txt_file_path, create_file_if_not_exists=True)
+            requirements_txt_file.set_requirement_and_write(requirement.to_python_package_requirement())
+        elif os.path.exists(requirements_txt_file_path):
+            requirements_txt_file = RequirementsTxtFile(requirements_txt_file_path)
+            requirements_txt_file.remove_requirement_and_write(requirement.to_python_package_requirement().key)
+
+    def has_module_requirement(self, module_name: str) -> bool:
```

Comment:
Why not just use `get_module_requirements()`? (`return any(module_name == Requirement.parse(req).key for req in self.get_module_requirements())`)

Reply:
Updated